velox/functions/prestosql/Arithmetic.h

```diff
@@ -69,6 +69,32 @@ struct MultiplyFunction {
   }
 };
 
+template <typename T>
+struct IntervalMultiplyFunction {
+  double sanitizeInput(double d) {
+    if UNLIKELY (std::isnan(d)) {
+      return 0;
+    }
+    return d;
+  }
+
+  template <typename T1, typename T2>
+  FOLLY_ALWAYS_INLINE void call(int64_t& result, const T1 a, const T2 b) {
```

Comment:
nit : Can you add some comments here saying if T1 is double the other type is not expected to be double etc, or just add some comments in general :)